### ISSUE TITLE
Sliding makes the entire UI freezing after update to NextJS 11.1.0

### ISSUE THREAD

**Issue by AdiMarianMutu** [davidcetinkaya/embla-carousel#209]
### Bug is related to

-  React 17.0.2 and NextJS 11.1.0

### Embla Carousel version

- v4.5.3

### Describe the bug

- When using `setSelectedIndex(embla.selectedScrollSnap())` to keep a state of the selected index, the entire UI slowly freezes, the more I slide the more it freezes.

### CodeSandbox

- I'm using the **exact same** code from [here](https://codesandbox.io/s/embla-carousel-thumbnails-react-hj33c)

### Steps to reproduce

- I want to specify that **before** updating to NextJS 11.1 everything worked just fine, so, after the update the first thing I've done it was to reinstall the previous NextJS version (I've completely deleted the .next and node_modules folder just to be sure) and to my surprise, the bug was still present!
I've spent more than 3 hours trying to understand what messed the carousel, so I've discovered that this specific part of the code makes everything freezing:

```
const onSelect = useCallback(() => {
    if (!embla || !emblaThumbs) return;

    setSelectedIndex(embla.selectedScrollSnap()); // <- HERE

    emblaThumbs.scrollTo(embla.selectedScrollSnap());
  }, [embla, emblaThumbs, setSelectedIndex]);
```

If I delete that line, the carousel does not freeze anymore and also the entire UI of my app works fine.
But, of course, I need that state in order to sync the thumb carousel with the main carousel.

### Expected behavior

- Not freezing

### Additional context

- Compare the smoothness of the carousel at the start with the smoothness at the end

[gif here](https://i.imgur.com/cV0Db57.mp4)

**Comment by AdiMarianMutu** [comment]
I've fixed the issue.

Inside the carousel I'm displaying the `Image` component from `next/image` with the new `onLoadingComplete` callback, this callback was creating a memory leak inside my callback function because I was trying to set a state on an unmounted component (but weird, because in the console I did not receive any message about trying to update an unmounted component).

So, the solution was to check if the component was still mounted. If the console would have given me that message I would have fixed this in a couple of minutes.

I hope this can be useful for someone who will face the same issue in the future.
